### ISSUE TITLE
Need promise-polyfill

### ISSUE THREAD

**Issue by evgeniy1801** [Foliotek/Croppie#226]
Croppie use native ES6 Promise, but some browsers not support this. Maybe need add promise-polyfill to dependensies?

**Comment by thedustinsmith** [comment]
https://github.com/Foliotek/Croppie/blob/master/croppie.js#L21

**Comment by evgeniy1801** [comment]
It's not work in IE11. After define polyfill in line 23 typeof Polyfill === 'undefined'.

**Comment by thedustinsmith** [comment]
I'm pretty sure it's working in IE11.  I'm not sure why you're doing `typeof Polyfill` ... I agree that that line will always equal undefined.

try `typeof Promise`...
![image](https://cloud.githubusercontent.com/assets/1330308/19600367/4ebfa7ec-976a-11e6-87e5-bb2cee8e352e.png)

**Comment by evgeniy1801** [comment]
Excuse me - i wrote incorrectly. Not `typeof Polyfill` - `typeof Promise`.
```
/* Polyfills */
    if (typeof Promise !== 'function') {
        /*! promise-polyfill 3.1.0 */
        !function(a){function b(a,b){return function(){a.apply(b,arguments)}}function c(a){if("object"!=typeof this)throw new TypeError("Promises must be constructed via new");if("function"!=typeof a)throw new TypeError("not a function");this._state=null,this._value=null,this._deferreds=[],i(a,b(e,this),b(f,this))}function d(a){var b=this;return null===this._state?void this._deferreds.push(a):void k(function(){var c=b._state?a.onFulfilled:a.onRejected;if(null===c)return void(b._state?a.resolve:a.reject)(b._value);var d;try{d=c(b._value)}catch(e){return void a.reject(e)}a.resolve(d)})}function e(a){try{if(a===this)throw new TypeError("A promise cannot be resolved with itself.");if(a&&("object"==typeof a||"function"==typeof a)){var c=a.then;if("function"==typeof c)return void i(b(c,a),b(e,this),b(f,this))}this._state=!0,this._value=a,g.call(this)}catch(d){f.call(this,d)}}function f(a){this._state=!1,this._value=a,g.call(this)}function g(){for(var a=0,b=this._deferreds.length;b>a;a++)d.call(this,this._deferreds[a]);this._deferreds=null}function h(a,b,c,d){this.onFulfilled="function"==typeof a?a:null,this.onRejected="function"==typeof b?b:null,this.resolve=c,this.reject=d}function i(a,b,c){var d=!1;try{a(function(a){d||(d=!0,b(a))},function(a){d||(d=!0,c(a))})}catch(e){if(d)return;d=!0,c(e)}}var j=setTimeout,k="function"==typeof setImmediate&&setImmediate||function(a){j(a,1)},l=Array.isArray||function(a){return"[object Array]"===Object.prototype.toString.call(a)};c.prototype["catch"]=function(a){return this.then(null,a)},c.prototype.then=function(a,b){var e=this;return new c(function(c,f){d.call(e,new h(a,b,c,f))})},c.all=function(){var a=Array.prototype.slice.call(1===arguments.length&&l(arguments[0])?arguments[0]:arguments);return new c(function(b,c){function d(f,g){try{if(g&&("object"==typeof g||"function"==typeof g)){var h=g.then;if("function"==typeof h)return void h.call(g,function(a){d(f,a)},c)}a[f]=g,0===--e&&b(a)}catch(i){c(i)}}if(0===a.length)return b([]);for(var e=a.length,f=0;f<a.length;f++)d(f,a[f])})},c.resolve=function(a){return a&&"object"==typeof a&&a.constructor===c?a:new c(function(b){b(a)})},c.reject=function(a){return new c(function(b,c){c(a)})},c.race=function(a){return new c(function(b,c){for(var d=0,e=a.length;e>d;d++)a[d].then(b,c)})},c._setImmediateFn=function(a){k=a},"undefined"!=typeof module&&module.exports?module.exports=c:a.Promise||(a.Promise=c)}(this);
        console.log(typeof Promise);
    }
```
I see `undefined` in ie11.

**Comment by thedustinsmith** [comment]
Wish I could help, but you see my screenshot. It's working for me.  Could you throw together a fiddle demonstrating the issue?

**Comment by dustinbrink** [comment]
I had the same problem, IE 11 was not defining Promise. I ended up moving the polyfill outside of the AMD for it to work correctly. My guess is, we have some scoping issue going on, loading Croppie via require.js

**Comment by imaximix** [comment]
I've been having the same problem.
But I was using webpack, and because of that the Promise would just get exported.
And the solution was pretty much the same as @dustinbrink 

tl;dr; polyfill promise yourself

**Comment by yevgen-grytsay** [comment]
This is because when you're useing webpack `this` isn't referencing `window`. This could help [SHIMMING MODULES](https://webpack.github.io/docs/shimming-modules.html#importing)

**Comment by ricdecastro2009** [comment]
I am a beginner with javascript and I am having difficulties. how did you solve this problem by loading croppie.js with require js?
It is necessary to add some new dependency?
